Add cron_schedule to StartWorkflowExecution

carlcortright opened the pull request:
We need a way to start cron workflows. This enables this functionality by adding it to the start workflow function in the client.

antstorm commented on lib/cadence.rb:
```diff
@@ -27,7 +27,8 @@ def start_workflow(workflow, *input, **args)
         workflow_id_reuse_policy: options[:workflow_id_reuse_policy],
-        headers: execution_options.headers
+        headers: execution_options.headers,
+        cronSchedule: execution_options.cron_schedule
```
^ underscore case here, for consistency :)

antstorm commented on examples/bin/load_test:
```diff
@@ -0,0 +1,19 @@
+  run_id = Cadence.start_workflow(workflow_class, *input, options: { workflow_id: workflow_id })
+  Cadence.logger.info "Started workflow: #{run_id} / #{workflow_id}"
+end
+
```
^ This file doesn't seem to be related to this PR

carlcortright replied:
Yeah not sure how this got in here... I'll remove it

antstorm commented on lib/cadence/execution_options.rb:
```diff
@@ -2,13 +2,15 @@
 module Cadence
   class ExecutionOptions
-    attr_reader :name, :domain, :task_list, :retry_policy, :timeouts, :headers
+    attr_reader :name, :domain, :task_list, :retry_policy, :timeouts, :headers,
```
ExecutionOptions are shared between workflows and activities, so `cron_schedule` probably doesn't belong here being workflow-specific concern

antstorm replied:
You can just reference it directly from the the options provided, similar to `workflow_id_reuse_policy`

antstorm reviewed (changes requested):
Thanks, Carl!

Left some inline comments, plus:
- I think it might be a good idea to have a separate API method for scheduling — `schedule_workflow` with `cron_schedule` being a top-level argument. WDYT?
- We'll also need an API for terminating a workflow, since that's the only way to stop a cron workflow

carlcortright commented:
@antstorm addressed your comments and also added the `schedule_workflow` method, I think that's a great idea. 

Thinking for termination we can split that into another PR. Thinking that we already have a few ways to cancel workflows through the UI and CLI. Would like to have a way to do it through ruby, but don't want to block this addition on that.

antstorm commented on spec/unit/lib/cadence_spec.rb:
```diff
@@ -72,6 +72,35 @@ class TestStartWorkflow < Cadence::Workflow
             )
         end
 
+        it 'starts a cron workflow' do
```
Can you please update the spec with the new `schedule_workflow` call?

antstorm commented on examples/bin/trigger_cron:
```diff
@@ -0,0 +1,19 @@
+
+run_id = Cadence.schedule_workflow(
+  workflow_class,
+  '* * * * *',
```
nit: Should we get the schedule string from command line args as well?

antstorm reviewed (approved):
🎉
